torch/csrc/dynamo/eval_frame.c

```diff
@@ -314,31 +396,228 @@ static void destroy_cache_entry(CacheEntry* e) {
   free(e);
 }
 
-inline static CacheEntry* get_cache_entry(PyCodeObject* code) {
-  CacheEntry* extra = NULL;
+typedef struct {
+  PyObject_HEAD
+  CacheEntry* cache_entry;
+} CacheEntryWrapper;
+
+static PyTypeObject CacheEntryWrapperType = {
+  PyVarObject_HEAD_INIT(NULL, 0)
+  .tp_name = "torch._C.dynamo.eval_frame.CacheEntryWrapper",
+  .tp_basicsize = sizeof(CacheEntryWrapper),
+  .tp_itemsize = 0,
+  .tp_flags = Py_TPFLAGS_DEFAULT,
+  .tp_new = PyType_GenericNew,
+};
+
+
+// The cache lives on the extra segment of code object. It can have one of the 4 possible values
+// 1) NULL - First time accessed the extra segment.
+// 2) SKIP_CODE - We will skip the original frame.
+// 3) CacheEntryWrapper - This contains the cache_entry for the frame.
+// 4) A dict from nn module to the CacheEntryWrapper.
```

Comment:
I hope it's not actually a dict, because if it is you are going to leak nn modules (since dict keys are strong references)